doc/source/dev/contributor/contributor_toc.rst

```diff
@@ -0,0 +1,76 @@
+.. _contributor-toc:
+
+=============================
+Contributor Table of Contents
+=============================
```

Comment:
New. Links to everything useful to new contributor that I could find. Linked from [`/source/index.rst`](https://github.com/scipy/scipy/pull/10286/files#diff-4bc0dbe7e164d778c3dc17f70c9f15ac)